### ISSUE TITLE
Platform TCK Messaging metadata version check is still broken due to com.sun.messaging.jms.ra.ConnectionMetaData in imqjmsra.jar returning 2.0 version numbers

### ISSUE THREAD

**Issue by scottmarlow** [eclipse-ee4j/glassfish#23253]
From testing with the GlassFish [October 22, 2020 (Thursday night) ](https://download.eclipse.org/ee4j/glassfish/glassfish-6.0.0-SNAPSHOT-nightly.zip) build, we are still getting Jakarta Messaging (JMS) Platform TCK failures due to `glassfish6/glassfish/lib/install/applications/jmsra/imqjmsra.jar` returning the wrong JMS version via `com.sun.messaging.jms.ra.ConnectionMetaData`.  

I verified that this occurs locally also by instrumenting the TCK test to show more information: `incorrect JMSMajorVersion=2 JmsTool.JMS_MAJOR_VERSION = 3 metadata cls= com.sun.messaging.jms.ra.DirectConnectionMetaData`

From inspecting a little closer, `imqjmsra/com/sun/messaging/jmq/jmsclient/ConnectionMetaDataImpl.class` ConnectionMetaData#getJMSMajorVersion returns `3` but `com.sun.messaging.jms.ra.ConnectionMetaData` (ConnectionMetaData#getJMSMajorVersion) returns `2`.

From disassembling com.sun.messaging.jms.ra.ConnectionMetaData (note that the `iconst_2` means `2` is returned):
```
public int getJMSMajorVersion() throws jakarta.jms.JMSException;
    descriptor: ()I
    flags: ACC_PUBLIC
    Code:
      stack=1, locals=1, args_size=1
         0: iconst_2
         1: ireturn
      LineNumberTable:
        line 87: 0
      LocalVariableTable:
        Start  Length  Slot  Name   Signature
            0       2     0  this   Lcom/sun/messaging/jms/ra/ConnectionMetaData;
    Exceptions:
      throws jakarta.jms.JMSException
```

The Jakarta EE 9 Platform TCK failure is shown at https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/master/1142/testReport/junit/com.sun.ts.tests.jms.core.topicConnection/TopicConnectionTests/jakartaeetck_run___jms_core_topicConnection___metaDataTests_from_ejb/
```
jakartaeetck-run / jms/core/topicConnection / com.sun.ts.tests.jms.core.topicConnection.TopicConnectionTests.metaDataTests_from_ejb (from jms_core_topicConnection) 
```
10-23-2020 09:54:23:  SVR-ERROR: Error: incorrect JMSMajorVersion=2
10-23-2020 09:54:23:  SVR-ERROR: Error: metaDataTests failed
10-23-2020 09:54:23:  SVR-ERROR: Test case throws exception: Error: metaDataTests failed
10-23-2020 09:54:23:  SVR-ERROR: Exception at:  
10-23-2020 09:54:23:  SVR-ERROR: com.sun.ts.lib.harness.EETest$Fault: Error: metaDataTests failed
	at com.sun.ts.tests.jms.core.topicConnection.TopicConnectionTests.metaDataTests(TopicConnectionTests.java:285)
```

**Comment by scottmarlow** [comment]
FYI, Alwin started an additional TCK test run via https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/master/1144 to confirm that the JMS metadata failure still occurs (in case the prior TCK test runs were incorrect in some way).

**Comment by pzygielo** [comment]
Indeed - it seems https://github.com/eclipse-ee4j/openmq/blob/eecc74c24f4775a1d26aa018d060faa82a92df39/mq/main/mq-jmsra/jmsra-ra/src/main/java/com/sun/messaging/jms/ra/ConnectionMetaData.java#L87 needs to be updated in OpenMQ.

**Comment by smillidge** [comment]
This need an update to OpenMQ which is integrated with GlassFish.

**Comment by edbratt** [comment]
Hi

I see a fix is proposed for OpenMQ. I believe we can wait a day or so to 
get this into GlassFish and verify everything is known good and then 
start the JMS ballot. I /think/ we have time in the schedule to get this 
started over the weekend or early next week.

Please make some noise when update TCK results are available.

-- Ed

**Comment by arjantijms** [comment]
> I see a fix is proposed for OpenMQ. I believe we can wait a day or so to
get this into GlassFish and verify everything is known good and then
start the JMS ballot.

To run the standalone TCK via Jenkins (as was done for many projects to derive the TCK Results from) a new build of GlassFish it not needed. The TCK job dynamically patches GlassFish with the desired OpenMQ version, which would be M3 here.

**Comment by smillidge** [comment]
Is this closed by #23254?

**Comment by scottmarlow** [comment]
https://ci.eclipse.org/jakartaee-tck/job/eftl-jakartaeetck-run-900/57/ passed the JMS tests which included running against GlassFish built via https://ci.eclipse.org/jakartaee-tck/job/build-glassfish/86/ that included the https://github.com/eclipse-ee4j/glassfish/pull/23254 fix.

We did hit a few typical stability issues that are not associated with the JMS issue.  We are rerunning the EJB tests which typically hit stability issues + the EL tests which I think also hit the stability issues.  The new TCK run with the EJB* + EL tests is https://ci.eclipse.org/jakartaee-tck/job/eftl-jakartaeetck-run-900/58/

I now consider that this JMS issue is verified as fixed and closing this issue as fixed by https://github.com/eclipse-ee4j/glassfish/pull/23254

If any further issues do show up (in EL/EJB tests which didn't yet run correctly with the JMS change), we will open a new issue for that.
